### PR TITLE
Clarify VisitHistory classes.

### PR DESCRIPTION
**Pull request by smillst**

### CODE REVIEW

File: framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java [hunk 1]
```diff
@@ -907,10 +912,17 @@ public Boolean visitWildcard_Wildcard(
     // interface but that handle cases that more than one visit method shares
     // in common
 
-    /** An intersection is a supertype if all of its bounds are a supertype of subtype. */
+    /**
+     * An intersection is a supertype if all of its bounds are a supertype of subtype.
+     *
+     * @param subtype the possible subtype
+     * @param supertype the possible supertype
+     * @return true if an intersection is a supertype if all of its bounds are a supertype of
```
Comment (mernst) [inline framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java:920]: I'm having trouble parsing this English.

File: framework/src/main/java/org/checkerframework/framework/type/StructuralEqualityComparer.java [hunk 2]
```diff
@@ -168,17 +168,21 @@ protected boolean areAllEqual(
 
     /**
      * First check visitHistory to see if type1 and type2 have been compared once already. If so
-     * return true; otherwise compare them and add them to visitHistory
+     * return true; otherwise compare them and put them in visitHistory
```
Comment (mernst) [inline framework/src/main/java/org/checkerframework/framework/type/StructuralEqualityComparer.java:171]: Nit: end sentence with period.

File: framework/src/main/java/org/checkerframework/framework/type/StructuralEqualityComparer.java [hunk 3]
```diff
@@ -168,17 +168,21 @@ protected boolean areAllEqual(
...
+     *
+     * @param type1 the first type
+     * @param type2 the second type
+     * @return whether the two types are equal
      */
     protected boolean checkOrAreEqual(
```
Comment (mernst) [inline framework/src/main/java/org/checkerframework/framework/type/StructuralEqualityComparer.java:177]: Not related to this pull request, but this method name is rather inelegant.

**Review by mernst — APPROVED** [review]
I have just a few small comments.

File: framework/src/main/java/org/checkerframework/framework/type/StructuralEqualityVisitHistory.java [hunk 4]
```diff
@@ -32,19 +32,25 @@ public StructuralEqualityVisitHistory() {
     /**
      * Add result of comparing {@code type1} and {@code type2} for structural equality for the given
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/type/StructuralEqualityVisitHistory.java:33]: Like in SubtypeVisitHistory, change the `Add` to `Put` to be consistent with the new method name?

**Pull request merged**